### PR TITLE
Routes A through D

### PR DESCRIPTION
**Pull request by cumins**
Completed the routes and testing for the following: 

- Reviews
- Categories
- Users (just testing the isAdmin attribute which we added to the model)
- Products

### CODE REVIEW

**Comment by cumins** [comment]
@parky22 @surabhinigam  - please code review our work when you can.

**Comment by runandrew** [comment]
Also, fix the ID and autoincrement bugs for each test

File: db/models/index.js [hunk 1]
```diff
@@ -21,11 +21,15 @@ const OrderProducts = require('./orderProducts');
...
 // Product foreign key on each Review
 Review.belongsTo(Product);
-Product.hasMany(Reviews);
+Product.hasMany(Review);
 
 // User foreign key on each Review
 Review.belongsTo(User, {as: 'Author'});
-User.hasMany(Review);
+
+//took the line below out, because we think it might not be necessary
```
Comment (parky22) [inline db/models/index.js:29]: agreed. we don't need this.

File: server/routes/categories.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+//DELETE a Category
+//api/Category/:categoryId
+Router.delete('/:categoryId', (req, res, next) => {
+	Category.destroy({
+		where: {
+			id: req.params.categoryId
+		}
+	})
+	.then((numDestroyedRows) => {
+		res.status(200).json(numDestroyedRows);
```
Comment (parky22) [inline server/routes/categories.js:38]: we need to make sure all delete status codes are consistent across routes. we'll go with this template.

File: server/routes/products.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+	})
+	.then((numDestroyedRows) => {
+		res.status(200).json(numDestroyedRows);
+	})
+	.catch(next);
+});
+
+//PUT (update) a product
+Router.put('/:productId', (req, res, next) => {
+	Product.update({ name: req.body.name,
```
Comment (parky22) [inline server/routes/products.js]: For the first argument in update, we might want to replace with req.body even though we discussed that there might be security issues. Omri pointed out to me and Surabhi that one benefit of doing just req.body is that we won't have to go through each of the routes to update specific fields if we decided to add or delete any later. He said we could then set up security measures elsewhere although I'm not sure how exactly.

File: server/routes/reviews.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+Router.get('/:reviewId', (req, res, next) => {
+	Review.findById(
+		req.params.reviewId
+	)
+	.then((singleReview) => {
+		res.json(singleReview);
+	})
+	.catch(next);
+});
+
```
Comment (parky22) [inline server/routes/reviews.js:51]: I think we should add at least 2 more GET routes for Review: 
1. '/api/reviews/product/:productId' - get all the reviews for one product
2. '/api/reviews/user/:userId' - get all the reviews written by one user (or author)
For these, we can probably use the get methods made through the associations we set up earlier right?

File: server/routes/reviews.test.js [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+describe('/api/reviews', () => {
+
+    const review1 = {
+        comment: 'No words',
+        rating: 5,
+        title: 'Andrews fabulous mustache'
+    };
+
+    const review2 = {
```
Comment (parky22) [inline server/routes/reviews.test.js]: literally crying XD

File: server/routes/reviews.test.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+            .then(res => expect(res.body).to.contain(review2))
+    );
+
+    it('GET all reviews...confirming we have 2 reviews', () =>
+        request(app)
+            .get(`/api/reviews`)
+            .expect(200)
+            .then(res => expect(res.body.length).to.equal(2))
+    );
+
```
Comment (parky22) [inline server/routes/reviews.test.js:66]: If we decided to add the 2 GET routes for review, then we should set up tests for them too.

File: server/routes/products.test.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+    const product2 = {
+        name: 'Joes fabulous mustache',
+        image: 'google.com/joesawesomemustache.jpg',
+        description: 'its really great',
+        quantity: 4,
+        price: 570.04
+    };
+
+	it('POST a new product(1)', () =>
```
Comment (parky22) [inline server/routes/products.test.js:25]: weird indent (not sure how to fix this on github directly sorry)

File: server/routes/reviews.test.js [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+        title: 'Andrews fabulous mustache'
+    };
+
+    const review2 = {
+        comment: 'I would buy 5 if I could',
+        rating: 5,
+        title: 'Joes exquisite mustache'
+    };
+
+	it('POST a new review(1)', () =>
```
Comment (parky22) [inline server/routes/reviews.test.js:44]: weird indent

File: server/routes/reviews.js [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+        res.json(updatedReview[1][0]); // Send back the updated review
+	})
+	.catch(next);
+});
+
+// POST - /api/reviews/ - posts a new review
+Router.post('/', (req, res, next) => {
+
+	Review.create({
+        comment: req.body.comment,
```
Comment (parky22) [inline server/routes/reviews.js:88]: going off of patterns through out, this should be indented more to align with the period before create.

File: server/routes/products.js (hunk 3 above)
Comment (surabhinigam) [reply server/routes/products.js]: yes I was trying to make the same point as well.

File: server/routes/reviews.js (hunk 4 above)
Comment (runandrew) [reply server/routes/reviews.js:51]: added

File: server/routes/products.js (hunk 3 above)
Comment (runandrew) [reply server/routes/products.js]: changed

**Pull request merged**